src/LuaInterface.cpp

```diff
@@ -342,43 +342,44 @@ bool LuaInterface::setCValue(QList<TVar*> vars)
     return false;
 }
 
+// sets the value of a Lua variable by running dynamically-generated Lua code
 bool LuaInterface::setValue(TVar* var)
 {
     //This function assumes the var has been modified and then called
     L = interpreter->pGlobalLua;
 
     QList<TVar*> vars = varOrder(var);
-    QString newName = vars[0]->getName();
+    QString variableChangeCode = vars[0]->getName();
     for (int i = 1; i < vars.size(); i++) {
         if (vars[i]->isReference()) {
             return setCValue(vars);
         }
         if (vars[i]->getKeyType() == LUA_TNUMBER) {
-            newName.append("[" + vars[i]->getName() + "]");
+            variableChangeCode.append("[" + vars[i]->getName() + "]");
         } else {
-            newName.append(R"([")" + vars[i]->getName() + R"("])");
+            variableChangeCode.append(R"([")" + vars[i]->getName() + R"("])");
```

Comment:
variableChangeCode.append(QStringLiteral(R"(["%1"])").arg(vars.at(i)->getName());

though I am **still** not sure that we can use the C++11 raw string literal when we use `tr(`...`)` anywhere within a compilation unit {due to, as you know [QTBUG-42736](https://bugreports.qt.io/browse/QTBUG-42736)} so may need to keep to:

    variableChangeCode.append(QStringLiteral("[\"%1\"]").arg(vars.at(i)->getName());